app/lib/page/reap_voucher/dialogs.dart

```diff
@@ -5,33 +5,65 @@ import 'package:encointer_wallet/store/app.dart';
 import 'package:encointer_wallet/utils/translations/index.dart';
 import 'package:flutter/cupertino.dart';
 
-Future<void> showRedeemSuccessDialog(BuildContext context) {
-  return showCupertinoDialog(
-    context: context,
-    builder: redeemSuccessDialog,
-  );
+class VoucherDialogs {
+  static Future<void> showRedeemSuccessDialog({
+    required BuildContext context,
+    required VoidCallback onOK,
+  }) {
+    final dic = I18n.of(context)!.translationsForLocale();
+
+    return showCupertinoDialog(
+      context: context,
+      barrierDismissible: true,
+      builder: (BuildContext context) {
+        return CupertinoAlertDialog(
+          title: const SizedBox(key: Key('voucher-dialog')),
+          content: Text(
+            dic.assets.redeemSuccess,
+          ),
+          actions: <Widget>[
+            CupertinoButton(
```

Comment:
Unfortunately, I couldn't test it myself. However, based on what you're saying, I think the error could be here.
```suggestions
         CupertinoButton(
              key: const Key('voucher_dialog_ok'),
              onPressed: onOK,
              child: Text(dic.home.ok),
              // onPressed: () {
              //   Navigator.of(context).popUntil((route) => route.isFirst);
              // },
            ),
```